lib/diagnostic.js

```diff
@@ -5,17 +5,33 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (previous, current) {
+const min = function (array, callback) {
+  const minimum = Math.min[array] // array.reduce(callback)
+  return minimum
+}
+//min([-1, -2, -3, -4, -5], callback(e, i, a))
 
+let min = function (array) {
+  let minimum = array[0]
+  for (let i = 1; i < array.length; i++) {
```

Comment:
This would work, but no need to use a `for loop` inside here since we're going to be passing this function to a `reduce` function.